odl/space/pspace.py

```diff
@@ -381,6 +384,107 @@ def dtype(self):
         else:
             raise AttributeError("`dtype`'s of subspaces not equal")
 
+    @property
+    def real_space(self):
+        """Variant of this space with real dtype.
+
+        Raises
+        ------
+        ValueError
+            Raised if the underlying spaces have different `dtype`, if `dtype`
+            is not a numeric data type, if `dtype` is not a real or complex
+            type, and if `dtype` is complex but cannot be mapped to a real
+            type.
+        """
+        try:
+            current_dtype = self.dtype
+        except(AttributeError):
+            raise ValueError(
+                'dtypes of underlying spaces are not the same and therefore '
```

Comment:
Well, technically I could imagine that a space of mixed complex64 and complex128 spaces would yield a space of mixed float32/64.

Wouldn't a easier implementation of all of this be to remove all logic and simply do:

```
real_spaces = [spc.real_space for doc in self]
return ProductSpace(*real_spaces)
```

Or does the logic add something to this?

Reply:
You are absolutely right. I think this is what's called bootstrapping, no? :-)